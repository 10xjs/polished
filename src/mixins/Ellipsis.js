/** @module Mixins */

/**
 * CSS to represent truncated text with an ellipsis.
 * @static
 * @param {string} [width = 100%] - Max-width for text to respect before being truncated.
 * @return {Object} CSS as JS.
 * @example
 * div {
 *   ellipsis(250px)
 * }
 *
 * // CSS as JS Output
 *
 * div {
 *   'display': 'inline-block',
 *   'max-width': '250px',
 *   'overflow': 'hidden',
 *   'text-overflow': 'ellipsis',
 *   'white-space': 'nowrap',
 *   'word-wrap': 'normal'
 * }
 */

function ellipsis(width = '100%') {
  return {
    'display': 'inline-block',
    'max-width': width,
    'overflow': 'hidden',
    'text-overflow': 'ellipsis',
    'white-space': 'nowrap',
    'word-wrap': 'normal',
  }
}

export default ellipsis
