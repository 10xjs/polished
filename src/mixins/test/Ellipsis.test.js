import Ellipsis from '../Ellipsis';

describe('Ellipsis', function() {

  it('should pass parameter to the value of max-width', function() {
    expect(Ellipsis('250px')).toMatchSnapshot();
  });

  it('should default max-width to 100%', function() {
    expect(Ellipsis()).toMatchSnapshot();
  });

});
