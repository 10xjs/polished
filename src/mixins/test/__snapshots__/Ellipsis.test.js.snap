exports[`Ellipsis should default max-width to 100% 1`] = `
Object {
  "display": "inline-block",
  "max-width": "100%",
  "overflow": "hidden",
  "text-overflow": "ellipsis",
  "white-space": "nowrap",
  "word-wrap": "normal",
}
`;

exports[`Ellipsis should pass parameter to the value of max-width 1`] = `
Object {
  "display": "inline-block",
  "max-width": "250px",
  "overflow": "hidden",
  "text-overflow": "ellipsis",
  "white-space": "nowrap",
  "word-wrap": "normal",
}
`;
